src/host/ncp_spinel.hpp

```diff
@@ -108,7 +108,8 @@ class NcpSpinel : public CliDaemon::Dependencies
     using InfraIfSendIcmp6NdCallback = std::function<void(uint32_t, const otIp6Address &, const uint8_t *, uint16_t)>;
     using BorderAgentMeshCoPServiceChangedCallback = std::function<void(bool, uint16_t, const uint8_t *, uint16_t)>;
     using CliDaemonOutputCallback                  = std::function<void(const char *)>;
-    using UdpForwardSendCallback = std::function<void(const uint8_t *, uint16_t, const otIp6Address &, uint16_t)>;
+    using UdpForwardSendCallback =
+        std::function<void(const uint8_t *, uint16_t, const otIp6Address &, uint16_t, uint16_t)>;
```

Comment:
```suggestion
        std::function<void(const uint8_t *, uint16_t, const otIp6Address &, uint16_t, const UdpForward &)>;
```

Reply:
I don't think we should use an object reference here. Because the callback is invoked from NcpSpinel when it receives the `UDP_FORWARD_STREAM`. So `NcpSpinel` doesn't have the access to the `UdpProxy` object. If setting the UdpProxy reference into `NcpSpinel`, it introduces extra dependency and initialization step, which seems unnecessary. Thoughts?